clang/include/clang/3C/ProgramInfo.h

```diff
@@ -68,7 +73,13 @@ class ProgramInfo : public ProgramVariableAdder {
   void exitCompilationUnit();
 
   bool hasPersistentConstraints(clang::Expr *E, ASTContext *C) const;
-  const CVarSet &getPersistentConstraints(clang::Expr *E, ASTContext *C) const;
+  const CSetBkeyPair &getPersistentConstraints(clang::Expr *E, ASTContext *C) const;
+  void storePersistentConstraints(clang::Expr *E, const CSetBkeyPair &Vars,
+                                  ASTContext *C);
+  // Get only constraint vars from the persistent contents of the
+  // expression E.
+  const CVarSet &getPersistentConstraintsSet(clang::Expr *E, ASTContext *C) const;
```

Comment:
This is a bit strange. The function has the same type signature as `getPersistentConstraints` but has a different name. How are the two functions different? Why do we need them both?